More robust conformer embed in QM module

xiaoruiDong opened the pull request:
This PR aims to partially solve issue #2138, by implementing a more robust conformed embedding scheme. The RDKit Default algorithm ETKDG may fail some time, which causes no conformers actually created in `rd_embed` functions that further raises an error when `GetConformer` at line 196.

The fix implements two backup algorithms to embed conformers: embedding using random coordinates, and embedding using 2D geometries. Since there is a following force field optimization step, it shouldn't be too bad an idea.

kspieks reviewed (changes requested):
Thanks for this PR that addresses the corresponding issue! The code works perfectly on my end for some test smiles. Can you please create a `moleculeTest.py` file that just tests the `rd_embed` method? Maybe test 1 or 2 smiles that ETKDG handles just fine, such as `CC` or any easy example. And also test 1 or 2 smiles, such as `[N:1](=[C-:2][C@@:3]1([H:8])[O:4][C@@:5]2([H:9])[C:6]([H:10])([H:11])[C+:7]12)[H:12]` and `[N+:1]1#[C:2][C@@:3]2([H:8])[O:4][C-:5]([H:9])[C:6]([H:10])([H:11])[C@@:7]12[H:12]`, that ETKDG fails to embed so that your new code for handling ETKDG failures can be tested as well.

xiaoruiDong commented on rmgpy/qm/molecule.py:
```diff
@@ -146,6 +146,29 @@ def rd_embed(self, rdmol, num_conf_attempts):
         AllChem.EmbedMultipleConfs(rdmol, num_conf_attempts, randomSeed=1)
 
+        if rdmol.GetNumConformers() == 0:
+            # There are molecules that the default ETKDG method fails.
```
Thanks. Corrected.

xiaoruiDong commented:
@kspieks  Thank you for reviewing this PR. I have
- corrected the wording
- added unit tests for `rd_embed`
- found rdkit force field optimization may fail some time, so I added a try...except... to catch the error. Since, only one conformed is needed in the end, it is okay to have some failures. I also added two flags to check the quality of embedding and optimization, but I diden't implement anything other than logging to prevent the calculation if both of them are bad. It requires some refactoring which is be beyond the scope of my original intention.

kspieks commented on rmgpy/qm/moleculeTest.py:
```diff
@@ -0,0 +1,87 @@
+
+import os
+import shutil
+import subprocess
```
It looks like `subprocess` is not used. Can you remove this import?

kspieks commented:
The new changes look great. Can you address the very minor comment about subprocess and then squash and rebase? This PR looks good to me :) Thanks again!

amarkpayne commented:
@xiaoruiDong can you go ahead and update this branch when you get a chance?

xiaoruiDong commented:
@amarkpayne Rebased the branch, thanks!
